torch/distributed/_tensor/ops/embedding_ops.py

```diff
@@ -115,7 +120,12 @@ def _reduce_shard_value(
         assert self.mask_buffer.data is not None
 
         # apply the mask to the tensor that pending reduction
-        tensor[self.mask_buffer.data, :] = 0.0
+        # NOTE: Depending on the use case (gather op or embedding op),
+        # tensor dim can be equal or larger than the mask dim, respectively.
+        if tensor.ndim == self.mask_buffer.data.ndim:
```

Comment:
given that we are reusing the logic, let's put this logic as a common method in MaskBuffer.